### DISCUSSION TITLE
Extracting gap profile

### DISCUSSION THREAD

**Discussion by sitangshugk95** [ContactEngineering/ContactMechanics#107] [Q&A]
For solving non-adhesive (hardwall) contact mechanics, is there a function that can be used to extract the local gap array after the solution has converged?

Using https://github.com/ContactEngineering/ContactMechanics/blob/eb97f877b5d3335a13f1d71b49be87d089247551/ContactMechanics/Systems.py#L124  after 
```
sol = nonadhesive_system.minimize_proxy(external_force=np.prod(t.physical_sizes) 
                                        * challenge_external_pressure, pentol=penetration_tol)

nonadhesive_disp = sol.x[0:nx,0:ny]
nonadhesive_offset = sol.offset
nonadhesive_gaps = nonadhesive_system.compute_gap(disp = nonadhesive_disp, offset = nonadhesive_offset)
```
spits out the following error: 

```
  File ~/anaconda3/lib/python3.11/site-packages/scipy/optimize/_optimize.py:225 in __getattr__
    raise AttributeError(name) from e

AttributeError: offset
```
Is this not compatible with the CCG optimizer? Any alternate ways to calculate it?

**Comment by sannant** [comment]
I think you should find `offset` in nonadhesive_system, not in `sol` . `sol` is the scipy minimization result object and knows nothing about the offset.
